silverback/_cli.py

```diff
@@ -102,6 +103,19 @@ def run(cli_ctx, account, runner_class, recorder_class, max_exceptions, path):
                 option_name="network", message="Network choice cannot support running app"
             )
 
+    if not path:
```

Comment:
I would say this is the order of processing that might work best:
1. `if path is None`: use `{cwd}/bot.py:bot` 
2. `if Path(path.split(":")[0]).exists()`: use `path[:{SilverbackApp instance varname}]`
3. `if not Path(path.split(":")[0]).exists() and "." not in path.split(":")[0]`: see if `{cwd}/bots/` exists, and `{path}.py`  exists in `{cwd}/bots/`, then use `{cwd}/bots/{path}.py[:{SilverbackApp instance varname}]`

Lastly, if no `:{SilverbackApp instance varname}` is present in path, assume `bot` (e.g. `bot = SilverbackApp()`)

---

But would say that we have some of this code already in `ape_run`, so might be useful to just lift (or copy) that processing here

Reply:
I'm looking at `ape_run` and it's pretty confusing. I get what we're doing there, but we already have this setup in `silverback run`:

We have `import_from_string(path)` in here that we're already using. I figured I'd just use that method after we create the path so I don't have to do any other work in here.

I can remove the `bots/` folder, since we talked about just running the runner from the `app/` folder. Otherwise, the functionality remains the same if you give it a path. Do you want me to read the file for the `SilverbackInstanceVarName = SilverbackApp()` to automatically set that variable? Should be easy to do with a grep

Reply:
> I'm looking at `ape_run` and it's pretty confusing. I get what we're doing there, but we already have this setup in `silverback run`:
> 
> We have `import_from_string(path)` in here that we're already using. I figured I'd just use that method after we create the path so I don't have to do any other work in here.

Yeah, that's a good point. That seems right. I think just pre-pend with `bots.` if you don't obtain one by trying it the first time and try again.

> I can remove the `bots/` folder, since we talked about just running the runner from the `app/` folder. Otherwise, the functionality remains the same if you give it a path. Do you want me to read the file for the `SilverbackInstanceVarName = SilverbackApp()` to automatically set that variable? Should be easy to do with a grep

No grep needed, either you set it explicitly (e.g. `:app`) or if you omit it we assume that it is `:bot` by adding that as a suffix (if `":"` not in the string)

Reply:
Sounds good, that's managed by the `import_from_string(path)`, if we want to add more checks to that, then I'll touch up that method. If `bot` doesn't exist in the module, we'll get a failure during runtime, which will be caught locally

Reply:
> if path is None: use {cwd}/bot.py:bot

Maybe we can make this more intelligent? Every single example in all the docs uses `app` for one. But also, maybe we should just check all the props of the module and see if they're of a `SilverbackApp` type?

Kind of hate the idea that you must have a file named X with a specific var named Y. Might be fine if we were generating a template project, where that was in a config somewhere. But it kind of gives me a bad taste to just assume everyone will structure their project in this way, which is in contrast to how most python projects are laid out. That (kind of) works for ape because the whole thing is structured around the idea of a project, but less so silverback.

Reply:
> Every single example in all the docs uses `app` for one. 

Yeah, this would need to be updated in the docs, and breaking change would be made

---

FastAPI has a similar sort of expectation they've created, if you have an `app.py` or `app/` module with `app` object in side it at the top level, and it will just run out of the box. Obviously there's a way to work around that, but that just serves to better standardize things for folks, especially for a relatively low importance decision of where and how to name your bots.

We changed from `app` to `bot` as the variable name because that might conflict with FastAPI

Reply:
Then I don't like how FastAPi does it either. Though supporting actual python modules seems a little more legit.  

And just updating the docs to enforce the idea we need a specific structure doesn't really improve how I feel about it.

Reply:
Yep, I'm aware at how it works. I don't really like it and I think it's going to be a real pain point for people building bots. But I'm not going to spend the day repeating the same thing. So I guess we'll just have to wait for user feedback.